e2e tests failed

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
When I run e2e tests (`./tests/e2e.sh`), I get an error:
```
* Done! kubectl is now configured to use "minikube"
! minikube IP was already correctly configured for 192.168.99.102
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T17:05:32Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
docker run --rm -v "$PWD":/go/src/k8s.io/kube-state-metrics -w /go/src/k8s.io/kube-state-metrics -e GOOS=linux -e GOARCH=amd64 -e CGO_ENABLED=0 golang:1.12 go build -ldflags "-s -w -X k8s.io/kube-state-metrics/pkg/version.Release=v1.6.0 -X k8s.io/kube-state-metrics/pkg/version.Commit=9a8e234 -X k8s.io/kube-state-metrics/pkg/version.BuildDate=2019-05-28T12:34:41Z" -o kube-state-metrics
can't load package: package k8s.io/kube-state-metrics: no Go files in /go/src/k8s.io/kube-state-metrics
Makefile:84: recipe for target '.container-amd64' failed
make: *** [.container-amd64] Error 1
```

**What you expected to happen**:
All e2e tests complete.

**How to reproduce it (as minimally and precisely as possible)**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T17:05:32Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```
- minikube version: `minikube version: v1.1.0`

- virtualbox version: `5.2.18_Ubuntu r123745`

- docker-machine version: `docker-machine version 0.16.0, build 702c267f`

I fixed the gopath and now it is such an:
```
# env | grep GOPATH
GOPATH=/home/user/go
```
I’m in the correct directory:
```
# pwd
/home/user/go/src/k8s.io/kube-state-metrics
```
But e2e tests continue to fall with the error:
```
* Done! kubectl is now configured to use "minikube"
! minikube IP was already correctly configured for 192.168.99.100
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T17:05:32Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
docker run --rm -v "$PWD":/go/src/k8s.io/kube-state-metrics -w /go/src/k8s.io/kube-state-metrics -e GOOS=linux -e GOARCH=amd64 -e CGO_ENABLED=0 golang:1.12 go build -ldflags "-s -w -X k8s.io/kube-state-metrics/pkg/version.Release=v1.6.0 -X k8s.io/kube-state-metrics/pkg/version.Commit=ce1bda0 -X k8s.io/kube-state-metrics/pkg/version.BuildDate=2019-05-29T16:13:08Z" -o kube-state-metrics
can't load package: package k8s.io/kube-state-metrics: no Go files in /go/src/k8s.io/kube-state-metrics
Makefile:84: recipe for target '.container-amd64' failed
make: *** [.container-amd64] Error 1
```

At the same time, manually launching `make container` works correctly:
```
docker run --rm -v "$PWD":/go/src/k8s.io/kube-state-metrics -w /go/src/k8s.io/kube-state-metrics -e GOOS=linux -e GOARCH=amd64 -e CGO_ENABLED=0 golang:1.12 go build -ldflags "-s -w -X k8s.io/kube-state-metrics/pkg/version.Release=v1.6.0 -X k8s.io/kube-state-metrics/pkg/version.Commit=ce1bda0 -X k8s.io/kube-state-metrics/pkg/version.BuildDate=2019-05-29T16:15:25Z" -o kube-state-metrics
cp -r * /tmp/tmp.771uhkN3Sr
docker build -t quay.io/coreos/kube-state-metrics-amd64:v1.6.0 /tmp/tmp.771uhkN3Sr
Sending build context to Docker daemon  53.73MB
Step 1/8 : FROM alpine:3.9
 ---> 055936d39205
Step 2/8 : RUN adduser -D kube-state-metrics
 ---> Using cache
 ---> 6a7541b1f3c9
Step 3/8 : FROM gcr.io/distroless/static
 ---> a574914f27fd
Step 4/8 : COPY kube-state-metrics /
 ---> f519f57d9e48
Step 5/8 : COPY --from=0 /etc/passwd /etc/passwd
 ---> 114e1da52d4b
Step 6/8 : USER kube-state-metrics
 ---> Running in 90fc3dcf18c2
Removing intermediate container 90fc3dcf18c2
 ---> a50c7f9c7b5b
Step 7/8 : ENTRYPOINT ["/kube-state-metrics", "--port=8080", "--telemetry-port=8081"]
 ---> Running in 903d5b5e50b7
Removing intermediate container 903d5b5e50b7
 ---> 4092a166dcec
Step 8/8 : EXPOSE 8080 8081
 ---> Running in bf6feee269f4
Removing intermediate container bf6feee269f4
 ---> 82f10d0ccc0f
Successfully built 82f10d0ccc0f
Successfully tagged quay.io/coreos/kube-state-metrics-amd64:v1.6.0
docker tag quay.io/coreos/kube-state-metrics-amd64:v1.6.0 quay.io/coreos/kube-state-metrics-amd64:latest
# Adding check for amd64
docker tag quay.io/coreos/kube-state-metrics-amd64:v1.6.0 quay.io/coreos/kube-state-metrics:v1.6.0
docker tag quay.io/coreos/kube-state-metrics-amd64:v1.6.0 quay.io/coreos/kube-state-metrics:latest
``

--- a/Makefile
+++ b/Makefile
@@ -5,6 +5,7 @@ TAG_PREFIX = v
 VERSION = $(shell cat VERSION)
 TAG = $(TAG_PREFIX)$(VERSION)
 LATEST_RELEASE_BRANCH := release-$(shell grep -ohE "[0-9]+.[0-9]+" VERSION)
+DOCKER_CLI ?= docker
 PKGS = $(shell go list ./... | grep -v /vendor/ | grep -v /tests/e2e)
 ARCH ?= $(shell go env GOARCH)
 BuildDate = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
@@ -61,14 +62,16 @@ doccheck: generate
 build-local: clean
 	GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(ARCH) CGO_ENABLED=0 go build -ldflags "-s -w -X ${PKG}/version.Release=${TAG} -X ${PKG}/version.Commit=${Commit} -X ${PKG}/version.BuildDate=${BuildDate}" -o kube-state-metrics
 
-build: clean
-	docker run --rm -v "${PWD}:/go/src/k8s.io/kube-state-metrics" -w /go/src/k8s.io/kube-state-metrics golang:${GO_VERSION} make build-local
+build: clean kube-state-metrics
+
+kube-state-metrics:
+	${DOCKER_CLI} run --rm -v "${PWD}:/go/src/k8s.io/kube-state-metrics" -w /go/src/k8s.io/kube-state-metrics golang:${GO_VERSION} make build-local
 
 test-unit: clean build
 	GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(ARCH) $(TESTENVVAR) go test --race $(FLAGS) $(PKGS)
 
 shellcheck:
-	docker run -v "${PWD}:/mnt" koalaman/shellcheck:stable $(shell find . -type f -name "*.sh" -not -path "*vendor*")
+	${DOCKER_CLI} run -v "${PWD}:/mnt" koalaman/shellcheck:stable $(shell find . -type f -name "*.sh" -not -path "*vendor*")
 
 # Runs benchmark tests on the current git ref and the last release and compares
 # the two.
@@ -91,26 +94,25 @@ all-container: $(addprefix sub-container-,$(ALL_ARCH))
 all-push: $(addprefix sub-push-,$(ALL_ARCH))
 
 container: .container-$(ARCH)
-.container-$(ARCH):
-	docker run --rm -v "${PWD}:/go/src/k8s.io/kube-state-metrics" -w /go/src/k8s.io/kube-state-metrics -e GOOS=linux -e GOARCH=$(ARCH) -e CGO_ENABLED=0 golang:${GO_VERSION} go build -ldflags "-s -w -X ${PKG}/version.Release=${TAG} -X ${PKG}/version.Commit=${Commit} -X ${PKG}/version.BuildDate=${BuildDate}" -o kube-state-metrics
+.container-$(ARCH): kube-state-metrics
 	cp -r * "${TEMP_DIR}"
-	docker build -t $(MULTI_ARCH_IMG):$(TAG) "${TEMP_DIR}"
-	docker tag $(MULTI_ARCH_IMG):$(TAG) $(MULTI_ARCH_IMG):latest
+	${DOCKER_CLI} build -t $(MULTI_ARCH_IMG):$(TAG) "${TEMP_DIR}"
+	${DOCKER_CLI} tag $(MULTI_ARCH_IMG):$(TAG) $(MULTI_ARCH_IMG):latest
 	rm -rf "${TEMP_DIR}"
 
 ifeq ($(ARCH), amd64)
 	# Adding check for amd64
-	docker tag $(MULTI_ARCH_IMG):$(TAG) $(IMAGE):$(TAG)
-	docker tag $(MULTI_ARCH_IMG):$(TAG) $(IMAGE):latest
+	${DOCKER_CLI} tag $(MULTI_ARCH_IMG):$(TAG) $(IMAGE):$(TAG)
+	${DOCKER_CLI} tag $(MULTI_ARCH_IMG):$(TAG) $(IMAGE):latest
 endif
 
 quay-push: .quay-push-$(ARCH)
 .quay-push-$(ARCH): .container-$(ARCH)
-	docker push $(MULTI_ARCH_IMG):$(TAG)
-	docker push $(MULTI_ARCH_IMG):latest
+	${DOCKER_CLI} push $(MULTI_ARCH_IMG):$(TAG)
+	${DOCKER_CLI} push $(MULTI_ARCH_IMG):latest
 ifeq ($(ARCH), amd64)
-	docker push $(IMAGE):$(TAG)
-	docker push $(IMAGE):latest
+	${DOCKER_CLI} push $(IMAGE):$(TAG)
+	${DOCKER_CLI} push $(IMAGE):latest
 endif
 
 push: .push-$(ARCH)

--- a/tests/e2e.sh
+++ b/tests/e2e.sh
@@ -27,6 +27,7 @@ E2E_SETUP_PROMTOOL=${E2E_SETUP_PROMTOOL:-}
 MINIKUBE_VERSION=v1.3.1
 MINIKUBE_DRIVER=${MINIKUBE_DRIVER:-virtualbox}
 SUDO=${SUDO:-}
+MINIKUBE_PROFILE=${MINIKUBE_PROFILE:-ksm-e2e}
 
 OS=$(uname -s | awk '{print tolower($0)}')
 OS=${OS:-linux}
@@ -77,9 +78,16 @@ mkdir "${HOME}"/.kube || true
 touch "${HOME}"/.kube/config
 
 export KUBECONFIG=$HOME/.kube/config
-${SUDO} minikube start --vm-driver="${MINIKUBE_DRIVER}" --kubernetes-version=${KUBERNETES_VERSION} --logtostderr
 
-minikube update-context
+if [[ "$MINIKUBE_DRIVER" != "none" ]]; then 
+  export MINIKUBE_PROFILE_ARG=" --profile ${MINIKUBE_PROFILE}"
+else
+  export MINIKUBE_PROFILE_ARG=''
+fi
+
+${SUDO} minikube start --vm-driver="${MINIKUBE_DRIVER}"${MINIKUBE_PROFILE_ARG} --kubernetes-version=${KUBERNETES_VERSION} --logtostderr
+
+minikube update-context${MINIKUBE_PROFILE_ARG}
 
 set +e
 
@@ -98,7 +106,7 @@ for _ in {1..90}; do # timeout for 3 minutes
 done
 
 if [[ ${is_kube_running} == "false" ]]; then
-   minikube logs
+   minikube logs${MINIKUBE_PROFILE_ARG}
    echo "Kubernetes does not start within 3 minutes"
    exit 1
 fi
@@ -107,8 +115,11 @@ set -e
 
 kubectl version
 
+# Build binary
+make build
+
 # ensure that we build docker image in minikube
-[[ "$MINIKUBE_DRIVER" != "none" ]] && eval "$(minikube docker-env)"
+[[ "$MINIKUBE_DRIVER" != "none" ]] && eval "$(minikube docker-env${MINIKUBE_PROFILE_ARG})" && export DOCKER_CLI='docker'
 
 # query kube-state-metrics image tag
 make container

